web-animations-1/Overview.bs

```diff
@@ -833,14 +833,58 @@ follows:
     <var>animation</var>, if any.
 1.  If <var>new timeline</var> is the same object as <var>old timeline</var>,
     abort this procedure.
+1.  Let |previous play state| be |animation|'s [=play state=].
+1.  Let |previous current time| be the |animation|'s [=current time=].
+1.  Let |had finite timeline| be true if |animation| has an associated
+    <a>timeline</a> that is not [=monotonically increasing=].
 1.  Let the <a>timeline</a> of <var>animation</var> be <var>new timeline</var>.
+1.  Perform the steps corresponding to the <em>first</em> matching
+    condition from the following, if any:
+
+    <div class="switch">
+
+    :   If |animation| has an associated
+        <a>timeline</a> that is not [=monotonically increasing=],
+
+    ::  1.  [=Apply any pending playback rate=] on |animation|
+        1.  Let |seek time| be zero if [=playback=] rate &ge; 0, and
+            |animation|'s <a>associated effect end</a> otherwise.
+        1.  Update the [=start time=] and [=hold time=] based on the first
+            matching condition if any
+
+            <div class="switch">
+
+            :   If <em>either</em> of the following conditions are true:
+                *    |previous playback rate| is 'running' or,
```

Comment:
I am wondering if it makes more sense to look at previous start/hold time instead of play states to determine what should be new start/hold time. The reason is that applying this logic on play/pause pending animations may introduce new states (see this [state chart](https://github.com/w3c/csswg-drafts/issues/2066#issuecomment-608095330))

Reply:
Switching the logic to rely on start and hold time, would also require factoring in the pending state.   Once pending state is introduced into the equation we essentially end up replicating the functionality of calculating the play state.  The new WPT tests associated with the Blink change test changing the timeline on both pending and non-pending animations.